### ISSUE TITLE
Build errors

### ISSUE THREAD

**Issue by impeeza** [Fledge68/WiiFlow_Lite#385]
On latest version of DevKitPro building Wiiflow_Lite you get the next set of errors:

![image](https://github.com/Fledge68/WiiFlow_Lite/assets/41979604/13b14bf1-cd54-4ac3-ab6c-43eea90ff3ab)

I am missing any library or dependency 

Even building a old commit which before I was able to build now I get the same set of errors.

**Comment by eku** [comment]
> On latest version

Please specify more precisely.

**Comment by Fledge68** [comment]
are you using  devkitppc r42-1 and libogc 2.4.0

**Comment by impeeza** [comment]
> are you using devkitppc r42-1 and libogc 2.4.0

No sir, I am Using:
MingW64 updated today


and the devkitpro packages:
- devkitpro-keyring	20180316-1
- devkitPPC		r45-1
- devkit-env		1.0.1-2
- libogc		2.6.0-1
- devkitppc-rules	1.2.1-1
- gamecube-tools	1.0.4-1
- general-tools		1.4.4-1

**Comment by Fledge68** [comment]
so if you would just use  devkitppc r42-1 and libogc 2.4.0 you shouldn't have those errors.

but as far as using the latest devkitppc and libogc im not sure why your libogc includes 

netinet/in.h

and  

sys/_sockaddr_storage.h

my libogc does not include those. i tried to see where they may have been added to libogc but couldn't find it.

obviously those files are conflicting the wolfssl files in portlibs/include/sys/socket.h

if you still want to use the latest ppc and libogc then goto portlibs/include/sys/socket.h and change all the socklen_t to socklen_t_w and sockaddr_storage to sockaddr_storage_w

and in source/libwolfssl/wolfio.h change the same ones to match socklen_t_w and sockaddr_storage_w

maybe that will work and maybe not.

**Comment by impeeza** [comment]
OK, first try, I did add this commands before build in order to install old versions of packages:

```
wget https://wii.leseratte10.de/devkitPro/devkitPPC/r42%20%282023-04-19%29/devkitPPC-r42.1-1-windows_x86_64.pkg.tar.zst
pacman -U --noconfirm "./devkitPPC-r42.1-1-windows_x86_64.pkg.tar.zst"

wget https://wii.leseratte10.de/devkitPro/libogc/libogc_2.4.0%20%282023-04-06%29/libogc-2.4.0-1-any.pkg.tar.zst
pacman -U --noconfirm "./libogc-2.4.0-1-any.pkg.tar.zst"
```

And then the process end fine, the boot.dol file was build without problem, only few warnnings:

```
{standard input}: Assembler messages:
{standard input}: Warning: end of file in comment; newline inserted
linking ... boot.elf
```
bur are cosmetic no functional.

Will try the tweaking of files now.

**Comment by impeeza** [comment]
Well after changing the files, using 

the next set of warnings appears to several files:
![image](https://github.com/Fledge68/WiiFlow_Lite/assets/41979604/cfaed1b1-9e00-43bb-b599-de6b730362b0)


and at end the next error interrupt the building:
![image](https://github.com/Fledge68/WiiFlow_Lite/assets/41979604/cfd7ba13-13c4-44fe-8353-e3ecb13eb631)


```
      |
T:/msys64/home/Impeeza/WiiFlow_Lite/source/network/https.c:306:5: error: conflicting types for 'connect'; have 'int(char *, u16)' {aka 'int(char *, short unsigned int)'}
  306 | int connect(char *host, u16 port)
      |     ^~~~~~~
T:/msys64/home/Impeeza/WiiFlow_Lite/portlibs/include/sys/socket.h:178:9: note: previous declaration of 'connect' with type 'int(int,  const struct sockaddr *, socklen_t_w)' {aka 'int(int,  const struct sockaddr *, unsigned char)'}
  178 | int     connect(int, const struct sockaddr *, socklen_t_w);
      |         ^~~~~~~
T:/msys64/home/Impeeza/WiiFlow_Lite/source/network/https.c: In function 'downloadfile':
T:/msys64/home/Impeeza/WiiFlow_Lite/source/network/https.c:374:25: warning: implicit declaration of function 'connect_W'; did you mean 'connect'? [-Wimplicit-function-declaration]
  374 |         httpinfo.sock = connect_W(getProxyAddress(), getProxyPort());
      |                         ^~~~~~~~~
      |                         connect
make[2]: *** [/opt/devkitpro/devkitPPC/base_rules:21: https.o] Error 1
make[1]: *** [/home/Impeeza/WiiFlow_Lite/Makefile.main:147: build] Error 2
make: *** [Makefile:12: all] Error 2

```

My C knowledge is very precarious so I can not help on this problem.

**Comment by impeeza** [comment]
Well after changing the files, using 

the next set of warnings appears to several files:
![image](https://github.com/Fledge68/WiiFlow_Lite/assets/41979604/cfaed1b1-9e00-43bb-b599-de6b730362b0)


and at end the next error interrupt the building:
![image](https://github.com/Fledge68/WiiFlow_Lite/assets/41979604/cfd7ba13-13c4-44fe-8353-e3ecb13eb631)


```
      |
T:/msys64/home/Impeeza/WiiFlow_Lite/source/network/https.c:306:5: error: conflicting types for 'connect'; have 'int(char *, u16)' {aka 'int(char *, short unsigned int)'}
  306 | int connect(char *host, u16 port)
      |     ^~~~~~~
T:/msys64/home/Impeeza/WiiFlow_Lite/portlibs/include/sys/socket.h:178:9: note: previous declaration of 'connect' with type 'int(int,  const struct sockaddr *, socklen_t_w)' {aka 'int(int,  const struct sockaddr *, unsigned char)'}
  178 | int     connect(int, const struct sockaddr *, socklen_t_w);
      |         ^~~~~~~
T:/msys64/home/Impeeza/WiiFlow_Lite/source/network/https.c: In function 'downloadfile':
T:/msys64/home/Impeeza/WiiFlow_Lite/source/network/https.c:374:25: warning: implicit declaration of function 'connect_W'; did you mean 'connect'? [-Wimplicit-function-declaration]
  374 |         httpinfo.sock = connect_W(getProxyAddress(), getProxyPort());
      |                         ^~~~~~~~~
      |                         connect
make[2]: *** [/opt/devkitpro/devkitPPC/base_rules:21: https.o] Error 1
make[1]: *** [/home/Impeeza/WiiFlow_Lite/Makefile.main:147: build] Error 2
make: *** [Makefile:12: all] Error 2

```

My C knowledge is very precarious so I can not help on this problem.

**Comment by JustMeDaFaq** [comment]
Hi, i guess it can be closed? Was able to compile it on latest devkitPPC and libogc.

devkitPPC: r46.1-1
libogc: 2.9.0-1

**Comment by Kirbeast** [comment]
I'm having trouble compiling the latest push d24a16e5c35cb4538ba7ac38c38d199757764a1a. I'm using devkitppc r42-1 and libogc 2.4.0. I get this build error `fatal error: gctypes.h: No such file or directory
    1 | #include <gctypes.h>
      |          ^~~~~~~~~~~
compilation terminated.`

**Comment by Kirbeast** [comment]
> Hi, i guess it can be closed? Was able to compile it on latest devkitPPC and libogc.
> 
> devkitPPC: r46.1-1 libogc: 2.9.0-1

Are you using windows 11? Did you have to do anything special to get it to compile?

**Comment by JustMeDaFaq** [comment]
> > Hi, i guess it can be closed? Was able to compile it on latest devkitPPC and libogc.
> > devkitPPC: r46.1-1 libogc: 2.9.0-1
> 
> Are you using windows 11? Did you have to do anything special to get it to compile?

Yeah im on Windows 11 and it works, (Linux worked too tho). While youre using the recommended devkitppc/libogc version i used the latest available at the time. How did you downgrade it? That might be an issue

**Comment by Kirbeast** [comment]
> Yeah im on Windows 11 and it works, (Linux worked too tho). While youre using the recommended devkitppc/libogc version i used the latest available at the time. How did you downgrade it? That might be an issue

To install devkitpro I used the graphic installer for windows. I downloaded the old version from the archive https://wii.leseratte10.de/devkitPro/. I extracted them. Renamed the current devkitPPC and libogc in C:\devkitpro. Copied the old version of the devkitPPC and libogc folders to C:\devkitpro.

**Comment by impeeza** [comment]
I personally am using MSYS2 on Windows 11.
